### PR TITLE
feat: (kernel) adds topic in schemadecorator

### PR DESCRIPTION
**Pull request by lordmallam**
Adds `topic_name` in `SchemaDecorator` as a JSONField defaulting to `{'name': [schemadeocrator.name]}`

Related JIRA Ticket: [AET-386](https://jira.ehealthafrica.org/browse/AET-386)

### CODE REVIEW

File: aether-kernel/aether/kernel/api/models.py [hunk 1]
```diff
@@ -414,6 +415,7 @@ class SchemaDecorator(ExportModelOperationsMixin('kernel_schemadecorator'), Proj
     transport_rule = models.TextField(null=True, blank=True, verbose_name=_('transport rule'))
     masked_fields = models.TextField(null=True, blank=True, verbose_name=_('masked fields'))
     is_encrypted = models.BooleanField(default=False, verbose_name=_('encrypted?'))
+    topic_name = JSONField(null=True, blank=True, verbose_name=_('topic name'))
```
Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/models.py]: I think that the field name should be just `topic`, shouldn't it?

Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/models.py]: Does it need to be a JSON field, could it be a simple TextField?

Comment (obdulia-losantos) [reply aether-kernel/aether/kernel/api/models.py]: @shawnsarwar ?

File: aether-kernel/aether/kernel/api/filters.py [hunk 2]
```diff
@@ -165,7 +165,7 @@ def mapping_filter(self, queryset, name, value):
             return queryset.filter(mappings__name=value)
 
     class Meta:
-        fields = '__all__'
+        exclude = ('topic',)
```
Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/filters.py:168]: Do we need to exclude the topic filed in filters? If we need it to be used in the producer it looks like that we need to filter the entries by this field...

Comment (obdulia-losantos) [reply aether-kernel/aether/kernel/api/filters.py:168]: :woman_shrugging:

File: aether-kernel/aether/kernel/api/migrations/0104_schemadecorator_topic_name.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Generated by Django 2.2.1 on 2019-05-15 09:57
+
+import django.contrib.postgres.fields.jsonb
+from django.db import migrations
+
+def migrate__update_schemadecorator(apps, schema_editor):
+    SchemaDecorator = apps.get_model('kernel', 'SchemaDecorator')
+    for sd in SchemaDecorator.objects.all():
+        sd.save() # default topic_name to { name: [schemadecorator.name] }
```
Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/migrations/0104_schemadecorator_topic_name.py]: flake8 warning... at least 2 spaces between the piece of code and the inline comment ;-)

File: aether-kernel/aether/kernel/api/models.py [hunk 4]
```diff
@@ -414,6 +415,7 @@ class SchemaDecorator(ExportModelOperationsMixin('kernel_schemadecorator'), Proj
     transport_rule = models.TextField(null=True, blank=True, verbose_name=_('transport rule'))
     masked_fields = models.TextField(null=True, blank=True, verbose_name=_('masked fields'))
     is_encrypted = models.BooleanField(default=False, verbose_name=_('encrypted?'))
+    topic = JSONField(null=True, blank=True, verbose_name=_('topic name'))
```
Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/models.py]: `verbose_name='topic'`

File: aether-kernel/aether/kernel/api/tests/test_models.py [hunk 5]
```diff
@@ -391,3 +391,18 @@ def test_model_deletion(self):
...
+        schema = models.Schema.objects.create(
+            name='schema',
+            definition=EXAMPLE_SCHEMA,
+        )
+        schemadecorator = models.SchemaDecorator.objects.create(
+            name='schema decorator',
+            project=project,
+            schema=schema,
+        )
+        self.assertEqual(schemadecorator.topic, {'name': 'schema decorator'})
```
Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/tests/test_models.py:408]: Could you add a couple of silly test in which you change the schemadecorator name and the topic doesn't change and the changing the topic one?

File: aether-kernel/aether/kernel/api/filters.py (hunk 2 above)
Comment (shawnsarwar) [reply aether-kernel/aether/kernel/api/filters.py:168]: The producer is going to use a single query that doesn't take topic into account. Don't need the filter for that specific purpose.

File: aether-kernel/aether/kernel/api/models.py (hunk 1 above)
Comment (shawnsarwar) [reply aether-kernel/aether/kernel/api/models.py]: Let's use a JSONField in case we want to put more advanced production options here.

**Review by obdulia-losantos — APPROVED** [review]

**Pull request merged**